kolibri/core/webpack/hooks.py

```diff
@@ -0,0 +1,290 @@
+            if hook.unique_slug == slug:
+                return hook
+        raise BundleNotFound("No bundle with that name is loaded: {}".format(slug))
+
+    @cached_property
+    @hooks.registered_method
+    def stats_file_content(self):
+        """
+        TODO: This property is only cached on the instance, maybe it should be
+        cached in a static module property instead so we can cache the JSON data
+        across the whole app?
+
+        :returns: A dict of the data contained in the JSON files which are
+        written by Webpack.
+        """
+        with open(self.stats_file) as f:
+            stats = json.load(f)
+        if django_settings.DEBUG:
+            timeout = 0
+            while stats['status'] == 'compiling':
+                time.sleep(getattr(settings, 'WEBPACK_POLL_INTERVAL', 0.1))
+                timeout += getattr(settings, 'WEBPACK_POLL_INTERVAL', 0.1)
+                with open(self.stats_file) as f:
+                    stats = json.load(f)
+                if timeout >= getattr(settings, 'WEBPACK_POLL_INTERVAL', 1.0):
+                    raise WebpackError('Webpack compilation still in progress')
+            if stats['status'] == 'error':
+                raise WebpackError('Webpack compilation has errored')
+        return {
+            "files": stats["chunks"][self.unique_slug]
+        }
+
+    @property
+    @hooks.registered_method
+    def bundle(self):
+        """
+        TODO: This is weird, why are we creating this as a dict object?
+
+        :returns: a generator yielding dict objects with properties of the built
+        asset, most notably its URL.
+        """
+        for f in self.stats_file_content["files"]:
+            filename = f['name']
+            if any(regex.match(filename) for regex in settings.IGNORE_PATTERNS):
+                continue
+            relpath = '{0}/{1}'.format(self.unique_slug, filename)
+            f['url'] = staticfiles_storage.url(relpath)
+            yield f
+
+    @hooks.registered_method
+    def bundle_filtered(self, extension=None):
+        """
+        TODO: Why is this helper function necessary!?
+
+        :returns: a possibly filtered list of data from self.bundle
+        """
+        bundle = self.bundle
+        if extension:
+            bundle = (chunk for chunk in bundle if chunk['name'].endswith('.{0}'.format(extension)))
+        return bundle
+
+    @property
+    @hooks.registered_method
+    def webpack_bundle_data(self):
+        """
+        This is the main interface to the NPM Webpack building util. It is
+        used by the webpack_json management command. Inheritors may wish to
+        customize this.
+
+        :returns: A dict with information expected by webpack parsing process.
+        """
+        return {
+            "name": self.unique_slug,
+            "src_file": self.src_file,
+            "static_dir": self.static_dir,
+            "stats_file": self.stats_file,
+            "events": {},
```

Comment:
This is incorrect - the events and once hashes need to be definable on everything.